### PR TITLE
Adding the integration test for Hadoop jobs

### PR DESCRIPTION
**Pull request by snleee**
1. Start MiniYarnMRCluster
2. Start Pinot cluster
3. Use hadoop job for pushing segments to Pinot
4. Run queries

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -355,7 +355,7 @@
       <dependency>
         <groupId>io.netty</groupId>
         <artifactId>netty</artifactId>
-        <version>3.10.5.Final</version>
+        <version>3.9.6.Final</version>
```
Comment (jenniferdai) [inline pom.xml]: I'm a little worried about this. We've had problems in the past where we have netty incompatibilities that only fail after 3 days or so of our cluster running prod. Can we try this maybe in lor1 perf?

File: pinot-hadoop/src/main/java/com/linkedin/pinot/hadoop/job/SegmentCreationJob.java [hunk 2]
```diff
@@ -202,7 +200,12 @@ public void run() throws Exception {
       throw new RuntimeException("Job failed : " + job);
     }
 
-    moveToOutputDirectory(fs);
+    try {
+      moveToOutputDirectory(fs);
+    } catch (Exception e) {
+      LOGGER.error("dsadsa", e);
```
[the remark this answers is not in the document]
Comment (snleee) [reply pinot-hadoop/src/main/java/com/linkedin/pinot/hadoop/job/SegmentCreationJob.java]: forgot to remove after debugging. removed.

File: pom.xml (hunk 1 above)
Comment (snleee) [reply pom.xml]: I checked that all of our production jars are pulling 3.9.6.Final

File: pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/HadoopSegmentPushOfflineClusterIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+    // Create the table
+    addOfflineTable(getTableName(), _schema.getTimeColumnName(), _schema.getOutgoingTimeUnit().toString(), null, null,
+        getLoadMode(), SegmentVersion.v3, getInvertedIndexColumns(), getTaskConfig(), _schema.getSchemaName());
+
+    // Generate and push Pinot segments from Hadoop
+    generateAndPushSegmentsFromHadoop();
+
+    // Wait for all documents loaded
+    waitForAllDocsLoaded(600_000L);
```
[the remark this answers is not in the document]
Comment (snleee) [reply pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/HadoopSegmentPushOfflineClusterIntegrationTest.java]: Starting Hadoop cluster is very slow. I will first let this in and if it causes transient errors, I will bump this number up.

File: pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/ClusterTest.java [hunk 4]
```diff
@@ -293,6 +293,7 @@ private static TableConfig getOfflineTableConfig(String tableName, String timeCo
         .setSegmentVersion(segmentVersion.toString())
         .setInvertedIndexColumns(invertedIndexColumns)
         .setTaskConfig(taskConfig)
+        .setSchemaName(schemaName)
```
[the remark this answers is not in the document]
Comment (snleee) [reply pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/ClusterTest.java]: removed

File: pinot-hadoop/src/main/java/com/linkedin/pinot/hadoop/job/JobConfigConstants.java [hunk 5]
```diff
@@ -27,6 +28,7 @@
   public static final String TABLE_PUSH_TYPE = "table.push.type";
   public static final String SCHEMA = "data.schema";
   public static final String SEGMENT_TABLE_NAME = "segment.table.name";
+  public static final String SEGMENT_NAME_POSTFIX = "segment.name.postfix";
```
[the remark this answers is not in the document]
Comment (snleee) [reply pinot-hadoop/src/main/java/com/linkedin/pinot/hadoop/job/JobConfigConstants.java]: As we discussed offline, I moved this to `HadoopSegmentCreationMapReduceJob`

File: pinot-hadoop/src/main/java/com/linkedin/pinot/hadoop/job/JobConfigConstants.java [hunk 6]
```diff
@@ -19,6 +19,7 @@
 public class JobConfigConstants {
   public static final String PATH_TO_INPUT = "path.to.input";
   public static final String PATH_TO_OUTPUT = "path.to.output";
+  public static final String PATH_TO_SCHEMA = "path.to.schema";
```
Comment (jenniferdai) [inline pinot-hadoop/src/main/java/com/linkedin/pinot/hadoop/job/JobConfigConstants.java]: Can we put a comment for this one that we are deprecating this and prefer that they fetch it from the controller?

File: pinot-hadoop/src/main/java/com/linkedin/pinot/hadoop/job/SegmentCreationJob.java [hunk 7]
```diff
@@ -211,29 +212,48 @@ public void run() throws Exception {
   }
 
   protected void setAdditionalJobProperties(Job job) throws Exception {
+    // Update push locations
```
Comment (jenniferdai) [inline pinot-hadoop/src/main/java/com/linkedin/pinot/hadoop/job/SegmentCreationJob.java]: Can we change this comment? Update seems like it's a different push location, maybe add?

File: pinot-hadoop/src/main/java/com/linkedin/pinot/hadoop/job/SegmentCreationJob.java [hunk 8]
```diff
@@ -211,29 +212,48 @@ public void run() throws Exception {
...
+
+    // Update table push type
+    job.getConfiguration().set(JobConfigConstants.TABLE_PUSH_TYPE, segmentPushType);
+
+    if (segmentPushType.equalsIgnoreCase(TableConfigConstants.APPEND)) {
+      // For append use cases, timeColumnName and timeType must be set
+      String timeColumnName = validationConfig.getTimeColumnName();
+      String timeColumnType = validationConfig.getTimeType();
+      if (timeColumnName == null || timeColumnType == null) {
+        throw new RuntimeException(
```
Comment (jenniferdai) [inline pinot-hadoop/src/main/java/com/linkedin/pinot/hadoop/job/SegmentCreationJob.java]: maybe add that those two are required for append use cases

**Pull request merged**